util/encoding/encoding.go

```diff
@@ -594,3 +595,30 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	}
 	return b, false
 }
+
+// EncodeTime encodes a time value, appends it to the supplied buffer,
+// and returns the final buffer. The encoding is guaranteed to be ordered
+// Such that if t1.Before(t2) then after EncodeTime(b1, t1), and
+// EncodeTime(b2, t1), Compare(b1, b2) < 0. The time zone offset not
+// included in the encoding.
+func EncodeTime(b []byte, t time.Time) []byte {
+	// Read the unix absolute time. This is the absolute time and is
+	// not time zone offset dependent.
+	sec := t.Unix()
+	nsec := t.Nanosecond()
+	if nsec < 0 {
+		// if it were negative we would need to change the way we encode.
```

Comment:
I think the panic was fine. `time.Time.nsec` must be in the range `[0, 999999999]` according to the docs. The `time` package seems to enforce this. 

Btw, if nsec were negative we would have to adjust it to be positive. Consider `sec=1, nsec=-1` vs `sec=0, nsec=999999999`. These are equivalent times yet encode to different values. And the `time.{Before,After}` comparisons would not work.